### PR TITLE
feat: Support `FOR SHARE`

### LINKED ISSUE #1878: [mysql] FOR SHARE
### Grammar or Syntax Description
- MYSQL
- Example: `FOR SHARE` clause is not supported yet

### SQL Example
Statement parse = CCJSqlParserUtil.parse(" SELECT * FROM MY_TABLE1 for share ");

### Additional context
jsqlparser-4.6.jar
mysql 8.0
[Links to the reference documentation](https://dev.mysql.com/doc/refman/8.0/en/innodb-locking-reads.html#innodb-locking-reads-nowait-skip-locked)
[SELECT ... FOR SHARE](https://dev.mysql.com/doc/refman/8.0/en/select.html)

### PR DESCRIPTION
**Pull request by jxnu-liguobin**
close #1878
MySQL: `FOR {UPDATE | SHARE}`
PostgreSQL: `FOR { UPDATE | NO KEY UPDATE | SHARE | KEY SHARE }`

### CODE REVIEW

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -2268,13 +2269,16 @@ PlainSelect PlainSelect() #PlainSelect:
 	[ LOOKAHEAD(<K_LIMIT>, { limit==null }) limit = LimitWithOffset() { plainSelect.setLimit(limit);    } ]
     [ LOOKAHEAD(<K_FETCH>) fetch = Fetch() { plainSelect.setFetch(fetch);    } ]
     [ LOOKAHEAD(<K_WITH> <K_ISOLATION>) withIsolation = WithIsolation() { plainSelect.setIsolation(withIsolation); } ]
-    [ LOOKAHEAD(2) <K_FOR> <K_UPDATE> { plainSelect.setForUpdate(true); }
+    [ LOOKAHEAD(2) <K_FOR> (
+            <K_UPDATE> { plainSelect.setForUpdate(true); } | <K_SHARE> { plainSelect.setForShare(true); } |
```
Comment (manticore-projects) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:2273]: Sorry to ask: are those types not Exlusive like:
- either `FOR UPDATE`
- or `FOR SHARE`
- or `FOR NO KEY UPDATE`
- or `FOR KEY SHARE`

If I am correct, was it not better to introduce a MODE based on ENUM values?
I do not like the possibility setting `ForUpdate` and `ForShare` and `ForNotKeyUpdate` and `ForKeyShare` to `TRUE` or `FALSE` all at the same time (via Java Code).

Also, please format the Grammar properly, its hard to read what is happening here.

File: src/test/java/net/sf/jsqlparser/statement/select/SelectTest.java [hunk 2]
```diff
@@ -4804,6 +4804,14 @@ public void testCurrentIssue940() throws JSQLParserException {
                 "SELECT date(current) AS test_date FROM systables WHERE tabid = 1");
     }
 
+    @Test
```
Comment (manticore-projects) [inline src/test/java/net/sf/jsqlparser/statement/select/SelectTest.java:4807]: Please add a test, where you set the modes using plain Java code, one after another.
I would assume, that setting one mode would have to switch off the other modes.

**Review by manticore-projects — CHANGES_REQUESTED** [review]
I have concerns on this implementation, when building this statement with java code.

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt (hunk 1 above)
Comment (jxnu-liguobin) [reply src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:2273]: 👍🏻  enumeration is a good way.

Comment (manticore-projects) [reply src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:2273]: You can check `java/net/sf/jsqlparser/statement/select/KSQLWindow.java` for the Enum `WindowType` to get a useful template.

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 3]
```diff
@@ -2268,13 +2269,19 @@ PlainSelect PlainSelect() #PlainSelect:
 	[ LOOKAHEAD(<K_LIMIT>, { limit==null }) limit = LimitWithOffset() { plainSelect.setLimit(limit);    } ]
     [ LOOKAHEAD(<K_FETCH>) fetch = Fetch() { plainSelect.setFetch(fetch);    } ]
     [ LOOKAHEAD(<K_WITH> <K_ISOLATION>) withIsolation = WithIsolation() { plainSelect.setIsolation(withIsolation); } ]
-    [ LOOKAHEAD(2) <K_FOR> <K_UPDATE> { plainSelect.setForUpdate(true); }
+    [ LOOKAHEAD(2)
```
Comment (manticore-projects) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:2272]: This looks much better, well done!

File: src/test/java/net/sf/jsqlparser/statement/select/SelectTest.java [hunk 4]
```diff
@@ -4804,6 +4805,27 @@ public void testCurrentIssue940() throws JSQLParserException {
...
+    public void testIssue1878() throws JSQLParserException {
+        assertSqlCanBeParsedAndDeparsed("SELECT * FROM MY_TABLE1 FOR SHARE");
+        // PostgreSQL ONLY
+        assertSqlCanBeParsedAndDeparsed("SELECT * FROM MY_TABLE1 FOR NO KEY UPDATE");
+        assertSqlCanBeParsedAndDeparsed("SELECT * FROM MY_TABLE1 FOR KEY SHARE");
+    }
+
+    @Test
+    public void testIssue1878ViaJava() throws JSQLParserException {
```
Comment (manticore-projects) [inline src/test/java/net/sf/jsqlparser/statement/select/SelectTest.java:4817]: Good!

**Review by manticore-projects — APPROVED** [review]

**Comment by manticore-projects** [comment]
Thank you a lot for providing this implementation!

**Pull request merged**